harvard-lts/drs-translation-service
# Handles unprocessed exports in the dropbox

**JIRA Issue**: https://jira.huit.harvard.edu/browse/LTSEPADD-31

# What does this Pull Request do?
Picks up any exports that were missed by the queuing system and processes them.

# How should this be tested?
Locally:

1. Clone the repo
2. Checkout the branch LTSEPADD-31
3. Copy the env-template.txt to .env and set necessary variables
4. Start up the docker container: `docker-compose -f docker-compose-local.yml up -d --build --force-recreate`
5. Exec into the container: `docker exec -it dts bash`
6. Run the command `pytest`

In dev:
This is already deployed to dev via the trial branch so to test that a batch gets picked up do the following.  Right now the unprocessed exports get picked up every 5 minutes.

1. Login to the Docker dev servers
2. Sudo into the `hdc3aadm` user
3. Rename the sample export with a unique ID `/docker/hdc3a/testdata/test_reprocess_adduniqueid` eg test_reprocess_<some timestamp>
4. Copy the renamed sample export to `/drs2dev/drsfs/dropboxes/epadddev_secure/incoming`
5. Verify that the batch gets picked up and a load report eventually ends up in `/drs2dev/drsfs/dropbox/lts_load_reports/epadddev_secure/incoming` (it should only take a 5 minutes or so)

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? Yes
- integration tests? No
